dis_snek/models/discord/channel.py

```diff
@@ -1773,25 +1779,55 @@ async def edit(
         Args:
             name: 1-100 character channel name
             archived: whether the thread is archived
+            applied_tags: list of tags to apply to a forum post (!!! This is for forum threads only)
             auto_archive_duration: duration in minutes to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080
             locked: whether the thread is locked; when a thread is locked, only users with MANAGE_THREADS can unarchive it
             rate_limit_per_user: amount of seconds a user has to wait before sending another message (0-21600)
             reason: The reason for this change
 
         Returns:
             The edited thread channel object.
-
         """
+        if applied_tags != MISSING:
+            applied_tags = [str(tag.id) if isinstance(tag, ThreadTag) else str(tag) for tag in applied_tags]
+
         return await super().edit(
             name=name,
             archived=archived,
             auto_archive_duration=auto_archive_duration,
+            applied_tags=applied_tags,
             locked=locked,
             rate_limit_per_user=rate_limit_per_user,
             reason=reason,
             **kwargs,
         )
 
+    @property
+    def applied_tags(self) -> list[ThreadTag]:
+        """
+        The tags applied to this thread.
+
+        Note:
+            This is only on forum threads.
+
+        """
+        if not isinstance(self.parent_channel, GuildForum):
+            raise AttributeError("This is only available on forum threads.")
+        return [tag for tag in self.parent_channel.available_tags if str(tag.id) in self._applied_tags]
+
+    @property
+    def initial_post(self) -> "Message":
+        """
+        The initial message posted by the OP.
+
+        Note:
+            This is only on forum threads.
+
+        """
+        if not isinstance(self.parent_channel, GuildForum):
+            raise AttributeError("This is only available on forum threads.")
+        return self.get_message(self.id)
+
 
 @define()
 class GuildPrivateThread(ThreadChannel):
```

Comment:
`ThreadChannel` is missing the field `flags` which is a integer bitshift flag to the `Channel`'s flags. This is new only for forum channels as well

Reply:
the `flags` field should be respected as a new type of object, `ChannelFlags` which stores only `1 << 1` for now which aliases to `PINNED`, when a forum post gets pinned to the top.